pkg/generated/applyconfiguration/climanager/v1/climanager.go

```diff
@@ -0,0 +1,248 @@
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/managedfields"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
+
+	climanager "github.com/openshift/cli-manager-operator/pkg/apis/climanager/v1"
+	"github.com/openshift/cli-manager-operator/pkg/generated/applyconfiguration/internal"
+)
+
+// CliManagerApplyConfiguration represents a declarative configuration of the CliManager type for use
+// with apply.
+type CliManagerApplyConfiguration struct {
+	v1.TypeMetaApplyConfiguration    `json:",inline"`
+	*v1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	Spec                             *CliManagerSpecApplyConfiguration   `json:"spec,omitempty"`
+	Status                           *CliManagerStatusApplyConfiguration `json:"status,omitempty"`
+}
+
+// CliManager constructs a declarative configuration of the CliManager type for use with
+// apply.
+func CliManager(name, namespace string) *CliManagerApplyConfiguration {
+	b := &CliManagerApplyConfiguration{}
+	b.WithName(name)
+	b.WithNamespace(namespace)
+	b.WithKind("CliManager")
+	b.WithAPIVersion("operator.openshift.io/v1")
+	return b
+}
+
+// ExtractCLIManager extracts the applied configuration owned by fieldManager from
+// cliManager. If no managedFields are found in cliManager for fieldManager, a
+// CliManagerApplyConfiguration is returned with only the Name, Namespace (if applicable),
+// APIVersion and Kind populated. It is possible that no managed fields were found for because other
+// field managers have taken ownership of all the fields previously owned by fieldManager, or because
+// the fieldManager never owned fields any fields.
+// cliManager must be a unmodified CliManager API object that was retrieved from the Kubernetes API.
+// ExtractCLIManager provides a way to perform a extract/modify-in-place/apply workflow.
+// Note that an extracted apply configuration will contain fewer fields than what the fieldManager previously
+// applied if another fieldManager has updated or force applied any of the previously applied fields.
+// Experimental!
+func ExtractCLIManager(cliManager *climanager.CliManager, fieldManager string) (*CliManagerApplyConfiguration, error) {
```

Comment:
I'm surprised to see that there is a trend towards implementing this utility function manually in auto generated applyconfiguration. For example https://github.com/openshift/client-go/blob/da6d55e4611fd99113270d4aa316ce48e71cac05/operator/applyconfigurations/operator/v1/cloudcredential.go#L33-L66
That also aligns with what you suggested.